.github/actions/action-setup_task-installPyProject/action.yml

```diff
@@ -0,0 +1,71 @@
+---
+# Adapted from @pvandyken's composite action
+name: Build and cache Python project
+description: Install project dependencies and optionally install project
+inputs:
+  python-version:
+    description: Python version to setup
+    required: true
+    type: string
+  reference:
```

Comment:
I'm actually pretty sure you can access the `github` context in a composite action. I'll try in my own snakebids version of this and let you know. But I think that's better than this vague `reference` input (sorry, I hadn't thought of this when I made my own push earlier

Reply:
Indeed it does work. Check out the latest commit there to see what I did

Reply:
I think you are right. I just ran into a context issue with the secrets, but that might just be because it is secrets and not inputs.

Reply:
Correct, secrets must be explicitly passed in (similar to inputs, but a different syntax)

Reply:
> Indeed it does work. [Check out the latest commit there to see what I did](https://github.com/akhanf/snakebids/pull/341/commits/c3f28d4fd3ddba4d0ce12444d4b2dea62b50d2b1)

Great, this change has been pushed in 8154b7d

> Correct, secrets must be explicitly passed in (similar to inputs, but a different syntax)

Yes, this gets a little confusing when chaining reusable workflows since a downstream workflow could inherit secrets from an upstream workflow.